src/mono/mono/utils/mono-threads-mach-helper.c

```diff
@@ -139,6 +154,24 @@ mono_threads_init_dead_letter (void)
 	id_objc_msgSend (pool, release);
 }
 
+static char *
+generate_partial_unique_classname ()
+{
+	guint8 partial_class_name [16];
+	gpointer rand_handle;
+	ERROR_DECL (error);
+
+	mono_rand_open ();
+	rand_handle = mono_rand_init (NULL, 0);
+
+	mono_rand_try_get_bytes (&rand_handle, (guint8*) partial_class_name, 16, error);
+	mono_error_assert_ok (error);
+
+	mono_rand_close (rand_handle);
+
+	return mono_guid_to_string_minimal ((guint8*) partial_class_name);
```

Comment:
The string returned form mono_guid_to_string_minimal needs to be freed when you are done using it using g_free.